src/utils/createElementFromRoutes.js

```diff
@@ -15,7 +15,9 @@ const createElementFromRoutes = ({
   }
   const element = matchedRoutes.reduceRight((childElement, route, matchedRouteIndex) => {
     const { component, components } = route;
-    if (typeof component !== 'function' && (!component || typeof components !== 'object')) {
+    const isComponent = ['object', 'function'].includes(typeof component);
```

Comment:
I unfortunately couldn't use `React.isValidElement` to check it since the component isn't rendered yet.